### PR TITLE
fix: send relative paths for external file system paths on Windows

### PR DESCRIPTION
**Pull request by paulRbr**
This commit is a bugfix for Windows based filesystem when external
references are defined on disk. Indeed we had a wrong regexp matchin
only unix style filesystem paths.

The fix consists of parsing windows style filesystem paths too.

I've tested it on a VM and the fix seems to work fine.

### CODE REVIEW

File: test/unit/definition.test.ts [hunk 1]
```diff
@@ -46,21 +46,14 @@ describe('API definition class', () => {
         const api = await API.load('examples/valid/asyncapi.v2.yml');
         expect(api.version).to.equal('2.2.0');
         expect(api.references.length).to.equal(5);
-        expect(api.references.map((ref) => ref.location)).to.include(
-          'http://example.org/param-lights.json',
-        );
+        const locations = api.references.map((ref) => ref.location);
```
Comment (scharrier) [inline test/unit/definition.test.ts:49]: Can we add a test for Windows path?

File: src/definition.ts [hunk 2]
```diff
@@ -108,8 +108,9 @@ class API {
     const refUrl = url(absPath);
 
     if (
-      absPath.match(/^\//) ||
-      (absPath.match(/^https?:\/\//) && definitionUrl.hostname === refUrl.hostname)
+      absPath.match(/^\//) || // Unix style filesystem path
+      absPath.match(/^[a-zA-Z]+\:\\/) || // Windows style filesystem path
+      (absPath.match(/^https?:\/\//) && definitionUrl.hostname === refUrl.hostname) // Same domain URLs
```
Comment (scharrier) [inline src/definition.ts:113]: Just a naive question: shouldn't we rely on `path.sep` instead, to explicitly identify/remove directory separators accordingly to the current OS?

**Review by scharrier — COMMENTED** [review]
Thanks for the fix, I let some questions :)

File: test/unit/definition.test.ts (hunk 1 above)
Comment (paulRbr) [reply test/unit/definition.test.ts:49]: I thought about it, but I remember I had troubles running tests on a windows worker. I'll try it again before merging

File: src/definition.ts (hunk 2 above)
Comment (paulRbr) [reply src/definition.ts:113]: I don't understand your question sorry. We don't remove directory separators (we delegate the relative path work to the `path` nodejs library) what we do is to detect if it's a filesystem path with the regexp

Comment (scharrier) [reply src/definition.ts:113]: 🤦 Sorry!

**Review by scharrier — APPROVED** [review]
Thanks for the fix, LGTM :+1:

**Pull request merged**